### PR TITLE
Eliminate connection presenters

### PR DESCRIPTION
**Pull request by jferris**
These presenters were largely dead code, as the ActiveModel pieces were
either unused or provided by the ActiveRecord objects they decorated.

Two methods were used:

* readable_name, which is replaced by i18n
* namespace, which was moved directly to the connection objects

I also renamed "namespace" to "integration_id" for greater parity with
routes and Rails conventions.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -159,6 +159,18 @@ en:
         We are exporting your profiles to NetSuite and will send you an email as
         soon as the export is complete.
 
+  greenhouse:
```
Comment (base10) [inline config/locales/en.yml]: You could also eliminate some duplication under connection_mailer.integration_mapping with this same approach.

Comment (jferris) [inline config/locales/en.yml]: I'm not finding the `integration_mapping` method. Is that merged yet?

Comment (base10) [inline config/locales/en.yml]: Sorry, that's at line 60 on `master`

Comment (jferris) [inline config/locales/en.yml]: Oh, I needed to rebase. Updated in a4132e0.

**Comment by base10** [comment]
LGTM

**Pull request merged**